rust/libnewsboat-ffi/src/utils.rs

```diff
@@ -531,3 +531,20 @@ pub unsafe extern "C" fn rs_strip_comments(line: *const c_char) -> *mut c_char {
         result.into_raw()
     })
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_extract_filter(line: *const c_char, filter: *mut c_char, url: *mut c_char) {
+    abort_on_panic(|| {
+        let line = CStr::from_ptr(line);
+        // `line` is a valid pointer from C string.
+        let line = line.to_str().expect("line contained invalid UTF-8");
+
+        let (rfilter, rurl) = utils::extract_filter(line);
+        // `rfilter` contains a subset of `line`, which is a C string. Thus, we conclude that
+        // `rfilter` doesn't contain null bytes. Therefore, `CString::new` always returns `Some`.
+        *filter = *CString::new(rfilter).unwrap().into_raw();
```

Comment:
This copies the first characters of `rfilter` into `filter`.

Just return a structure with two `char*` fields, that'd be much easier to get right.

Reply:
That is true but the structure is only used here so it will be wasted. If we were to use this method, how can it be done correctly?

Reply:
> That is true but the structure is only used here so it will be wasted.

That's true, but: 1) you have trouble handling pointers, so picking the simplest solution (a structure) should help; 2) these bindings will be removed sometime in the future, so it doesn't make sense to put too much engineering effort into them.

> If we were to use this method, how can it be done correctly?

1. Declare a structure on the C++ side:
    ```c++
    struct X { char* filter; char* url }
    ```
2. Make this function return that structure
3. Declare the same structure on Rust side (don't forget `repr[C]`)
4. Make Rust bindings re-pack the structure into a tuple

Reply:
> 
    Declare a structure on the C++ side:

    struct X { char* filter; char* url }

    Make this function return that structure
    Declare the same structure on Rust side (don't forget repr[C])
    Make Rust bindings re-pack the structure into a tuple

I was asking for the other method but never mind, this is just temporary.

> That's true, but: 1) you have trouble handling pointers, so picking the simplest solution (a structure) should help; 2) these bindings will be removed sometime in the future, so it doesn't make sense to put too much engineering effort into them.

True